syntax = "proto3";

package pki;

import "google/protobuf/empty.proto";

service Register {
  rpc CreateIdentity(Identity) returns (Certificate) {}
  rpc VerifyCertificate(Certificate) returns (google.protobuf.Empty) {}
}

message Identity {
  string hostname = 1;
  string country = 2;
  string state = 3;
  string org = 4;

}

message Certificate {
  string server_pubkey = 1;
  string server_privkey = 2;
  string server_cert = 3;
}